### ISSUE TITLE
Preprocessing

### ISSUE THREAD

**Issue by cris-pixel** [phuycke/ML-in-Python#2]
Hello @MaartenVelghe and @danieltomasz 

For the preprocessing, give it a check just to see if it is right (i based myself on the solution and the notebooks for the practica) these are the points that I struggled with:

1. - Rescaling: That part didn't work. It is because of the dummification. Instead of making the missing data a separate category, I just used the mode.... but I did not change the gender to "0" and "1" but let them as "M" and "F" (the training and score tests are fused, I added the last step to "unfuse them).
2. - Identify outliers, and choose whether or not to make your analysis more robust by removing these: I decided to continue using them...
3. - I don't know if you would like to add some graphics to survey the data...
4. 22/06/2021===> I added something to the code and tried to "push" the changes...

**Comment by danieltomasz** [comment]
@cris-pixel  I made some small changes  to your script to be able to run it on my computer, I will try to work on imputing the data and visualization, I also put the file into the 'pipeline' folder

@cris-pixel  I also recommend syntax
```
# %%  some comment marking new chunk of code
```
over
```
# %%some comment marking new chunk of code
```
The space between `%%` and comments allows me to convert script into Jupyter notebook using 
```
jupytext --to notebook notebook.py  
```
(but only after I installed jupytext to my environment)

**Comment by cris-pixel** [comment]
Ahhh!!! Ok! I had no idea that that could be done. I will use it for the following parts of my code. I am working on question 2.

**Comment by danieltomasz** [comment]
@cris-pixel  you could also easily convert notebooks into py files using jupytext
https://jupytext.readthedocs.io/en/latest/using-cli.html

**Comment by cris-pixel** [comment]
Hello @danieltomasz  and @MaartenVelghe,
This is what I have until now from my question. I will work on it during the weekend (basically i copied parts of the practice), I think I just need to comment it and fit it.

# %% Separate the test and the train sets
data_train = pd.concat([data_train[['outcome_profit', 'outcome_damage_inc', 'outcome_damage_amount']],datafull2[0:5000]], axis=1)
print(data_train.shape)
score = datafull2[5000:5500]
score.shape

# in here we are spliting the train and the test data sets. We decided to use 20% for testing and 80% for training
X_train, X_test, y_train, y_test = train_test_split(data_train.drop(['outcome_profit', 'outcome_damage_inc', 'outcome_damage_amount'],1),  
                                                    data_train['outcome_profit'], test_size=0.2, random_state=42)

# Then we are going to use Gradient boost
n_estimators = [int(x) for x in np.linspace(start = 100, stop = 1000, num = 50)]
learning_rate = [x for x in np.logspace(start = -3, stop = -0.01, num = 50)]
max_features = ['auto']
max_depth = [int(x) for x in np.linspace(1, 10, num = 10)]
min_samples_split = [2, 5, 10, 30]
min_samples_leaf = [1, 2, 4, 10, 30]
subsample = [0.4, 0.6, 0.8, 1]
random_grid = {'n_estimators': n_estimators,
               'learning_rate': learning_rate,
               'max_features': max_features,
               'max_depth': max_depth,
               'min_samples_split': min_samples_split,
               'min_samples_leaf': min_samples_leaf,
               'subsample': subsample}
gbm = GradientBoostingRegressor()
gbm_random = RandomizedSearchCV(estimator = gbm, param_distributions = random_grid, n_iter = 200, cv = 5, verbose=2, random_state=48, n_jobs = -1)
gbm_random.fit(X_train, y_train)
gbm_random.best_params_

#%% In this part, we are fitting the model for PROFIT
params = gbm_random.best_params_
gbm_profit = GradientBoostingRegressor(**params)
gbm_profit.fit(X_train, y_train)
print('R2: %.3f' % gbm_profit.score(X_train, np.array(y_train).reshape(-1,1)))
print('R2: %.3f' % gbm_profit.score(X_test, np.array(y_test).reshape(-1,1)))

#%% And now we are scoring the 500 potential customers with it
profit_preds = gbm_profit.predict(score)

**Comment by danieltomasz** [comment]
I will be working on the dataset today evening, I will let both of you know when I will push my updates

**Comment by cris-pixel** [comment]
Hello @danieltomasz , I worked on the preprocessing too (just on the part that I had mistakes), I think now it runs.... I also finished question 2, and added something of "white boxing" I will upload my file.

**Comment by danieltomasz** [comment]
Nice! Remember  in terminal
```
git add .
git commit  -m "your message"
git pull 
git push
``
You could  in GUI client too

**Comment by danieltomasz** [comment]
@cris-pixel  I will try to visualise some of the the data, I am working on my copy of your script, after you uploads yours we migjht pick what is best for the final report/script

**Comment by cris-pixel** [comment]
Thank you!, I will do what you put in your message (I uploaded the files, but not in that way).

**Comment by danieltomasz** [comment]
The files aren't uploaded, you may try to do it through github interface if you have problem with git

there is nice plugin for spyder for use of notebooks
https://github.com/spyder-ide/spyder-notebook

@cris-pixel  I may show you tomorrow how to use Jupyter & how to work with git, I am going to sleep

**Comment by cris-pixel** [comment]
Thank you!
